sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -57,7 +60,7 @@ case class CreateDatabase(
       CatalogDatabase(
         databaseName,
         comment.getOrElse(""),
-        path.getOrElse(catalog.getDefaultDBPath(databaseName)),
+        catalog.createDatabasePath(databaseName, path),
```

Comment:
yeah, actually how is `new Path(p).toString` different from `p`? Looks like that is the only functional difference here? If that doesn't do anything then let's just revert all these changes here and only commit the test changes.